app/helpers/samples_helper.rb

```diff
@@ -196,6 +215,40 @@ def get_total_runtime(pipeline_run, run_stages)
     end
   end
 
+  def filter_by_time(samples, start_date, end_date)
+    samples.where("samples.created_at >= ? AND samples.created_at <= ?", start_date, end_date)
+  end
+
+  def filter_by_visibility(samples, visibility)
+    if visibility
+      public = visibility.include?("public")
+      private = visibility.include?("private")
+
+      if public ^ private
```

Comment:
why XOR here? why not `elsif private` ? flat is better than nested

Reply:
Because that block only applies if they exclusively selected (not simultaneously).